Fix improper `class.filename` and `package.name` values on XML coverage report

lormico opened the pull request:
Solves #862

arcivanov commented:
flake8 warnings. @lormico

lormico commented:
Whoooops, fixed it

arcivanov commented:
@lormico failures on Windows.

https://github.com/pybuilder/pybuilder/runs/7585801888?check_suite_focus=true#step:3:1471

lormico commented:
It looks like the Windows failures are not caused by my edit on `coverage_plugin.py`: I tried [reverting my changes and disabling the assertions in the integration test](https://github.com/lormico/pybuilder/commit/e6546b6f95ee759dffd053317787ed849171d2f8), and it [still fails](https://github.com/lormico/pybuilder/runs/7591952345?check_suite_focus=true) when running the `coverage` step.

Maybe there's something wrong with GitHub's Windows testing platform? Should I make PyBuilder skip my test if `"win" in sys.platform`?

arcivanov commented:
No, it is not. It's the issue with disks/locations.

> ValueError: path is on mount 'c:', start on mount 'D:'

arcivanov commented:
@lormico it's the change related to path manipulation. Current tree works fine:

https://github.com/pybuilder/pybuilder/runs/7593136238?check_suite_focus=true

lormico commented:
> No, it is not. It's the issue with disks/locations.
> 
> > ValueError: path is on mount 'c:', start on mount 'D:'

I searched a bit for this issue, and it seems like it's due to how Python itself handles relative paths on Windows, when two different volumes are involved. I'm still not completely sure about how this is connected to 

Sorry, I'm not sure I understand what's going on here. 

I keep getting the [same error](https://github.com/lormico/pybuilder/runs/7614519612?check_suite_focus=true), even if the `src/main` codebase is the same as the one on `pybuilder/master`.

I'm unfortunately not very experienced in GitHub Actions, do you have any pointers on what could be happening?

lormico commented:
Ok, I wanted to investigate deeper on this issue, so I created a branch on `lormico/pybuilder` from the latest commit on `pybuilder/pybuilder/master` and:
1. removed all integration tests (merely for build speedup)
2. added a [new integration test](https://github.com/lormico/pybuilder/blob/test_coverage_failure_on_windows/src/integrationtest/python/should_not_fail_coverage_tests.py) which only sets up a simple project using the methods provided by `itest_support.IntegrationTestSupport` and then calls the `coverage` task (no assertions)

I still keep getting the [same error](https://github.com/lormico/pybuilder/runs/7614519612?check_suite_focus=true), even if the `src/main` codebase is the same as the one on `pybuilder/master`, so I'd exclude that it's because of something that was introduced recently.

This particular error:
> ValueError: path is on mount 'c:', start on mount 'D:'

seems to be due to how Python itself handles relative paths on Windows, when two different volumes are involved ([here](https://bugs.python.org/issue7195) a very old ticket from 2009 which explains this better than I could do). 

I'm still not completely sure about how this is connected to PyBuilder, but I'm more inclined to think that's because of how GitHub Actions work or how they are configured for a Windows VM. Alas, I have nearly zero experience with GitHub Actions, so I might be off track. 

Do you have any pointers from here @arcivanov?

arcivanov commented:
@lormico I took another dig at it and it looks like a bug in coverage

nedbat/coveragepy#1428

arcivanov commented:
@lormico Please rebase on the the latest. The test also needs to be modified to not fail for unrelated reasons:

```
       File "d:\a\pybuilder\pybuilder\target\dist\pybuilder-0.13.7.dev\pybuilder\execution.py", line 95, in execute
        self.callable(*arguments)
        ^^^^^^^^^^^^^^^^^^^^^^^^^
      File "d:\a\pybuilder\pybuilder\target\dist\pybuilder-0.13.7.dev\pybuilder\plugins\python\coverage_plugin.py", line 157, in coverage
        raise failure
        ^^^^^^^^^^^^^
    pybuilder.errors.BuildFailedException: Test coverage for at least one module is below 70%
    
    ----------------------------------------------------------------------
```

Very close now :smile:

lormico commented:
I ran the Actions on my repo (I had to disable the `upload` step for `pyb` in order to make it run successfully, because of permissions), but during the build PyBuilder seems to have downloaded an old version of `coverage`, and failed on these two instances: [[windows-latest, 3.11.0-beta.5, true, false]](https://github.com/lormico/pybuilder/runs/7711519920?check_suite_focus=true) and [[windows-latest, 3.11.0-beta.5, false, false]](https://github.com/lormico/pybuilder/runs/7711519980?check_suite_focus=true).

The error reveals it is not using the latest version:
```python
message(f"Combined data file {os.path.relpath(f)}")  # should be message(f"Combined data file {file_name}")
```
even though the dependency specification in `pyb` is `coverage~=6.0`, which should be ok.

I'm doing some tests to find out what is the cause.

lormico commented:
Nevermind, it now doesn't seem to fail anymore.

@arcivanov can you please approve launching the workflow?

arcivanov commented:
@lormico Thank you, merged.
